src/components/AddPresetAsset.tsx

```diff
@@ -0,0 +1,180 @@
+import { getPresetAssets } from "helpers/getPresetAssets";
+import { getValidatedUntrustedAsset } from "helpers/getValidatedUntrustedAsset";
+import { log } from "helpers/log";
+import { searchParam } from "helpers/searchParam";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus, SearchParams } from "types/types.d";
+import { shortenStellarKey } from "helpers/shortenStellarKey";
+
+export const AddPresetAsset = ({ onClose }: { onClose: () => void }) => {
+  const { account, allAssets, settings, untrustedAssets } = useRedux(
+    "account",
+    "allAssets",
+    "settings",
+    "untrustedAssets",
+  );
+  const [presetAssets, setPresetAssets] = useState<presetAsset[]>([]);
+  const [checkedAssets, setCheckedAssets] = useState<{
+    [key: string]: boolean;
+  }>({});
+  const [isValidating, setIsValidating] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const history = useHistory();
+
+  useEffect(() => {
+    setPresetAssets(getPresetAssets(allAssets.data));
+  }, [allAssets]);
+
+  useEffect(() => {
+    if (untrustedAssets.status === ActionStatus.SUCCESS) {
+      onClose();
+    }
+
+    if (untrustedAssets.errorString) {
+      setErrorMessage(untrustedAssets.errorString);
+    }
+  }, [onClose, untrustedAssets.errorString, untrustedAssets.status]);
+
+  const getAssetId = (asset: presetAsset) =>
+    `${asset.assetCode}:${asset.homeDomain || asset.issuerPublicKey}`;
+
+  const handleOnChangeCheckbox = (asset: presetAsset) => {
+    const updatedCheckedAssets = JSON.parse(JSON.stringify(checkedAssets));
+    const assetId = getAssetId(asset);
+    updatedCheckedAssets[assetId] = !checkedAssets[assetId];
+    setCheckedAssets(updatedCheckedAssets);
+  };
+
+  const handleAddUntrustedAssets = async (assetList: presetAsset[]) => {
+    setErrorMessage("");
+    setIsValidating(true);
+
+    try {
+      const validatedAssetValues = await Promise.all(
+        assetList.map(async (pAsset) => {
+          const { assetCode, issuerPublicKey, homeDomain } = pAsset;
+
+          if (!(homeDomain || issuerPublicKey)) {
+            const errorMsg = `Home domain OR issuer public key is required with asset code ${assetCode}`;
+            log.error({ title: errorMsg });
+            throw new Error(errorMsg);
+          }
+
+          const asset = await getValidatedUntrustedAsset({
+            assetCode,
+            homeDomain,
+            issuerPublicKey,
+            accountBalances: account.data?.balances,
+            networkUrl: getNetworkConfig(settings.pubnet).url,
+          });
+
+          return `${asset.assetCode}:${asset.assetIssuer}`;
+        }),
+      );
+
+      const newSearchQ = searchParam.update(
+        SearchParams.UNTRUSTED_ASSETS,
+        validatedAssetValues,
+      );
+      history.push(newSearchQ);
+    } catch (e) {
+      const errorMsg = getErrorMessage(e);
+      log.error({ title: errorMsg });
+      setErrorMessage(errorMsg);
```

Comment:
If one of the assets fail, we quit the whole thing, correct?

Reply:
Yes, that's correct. The risk is low, considering those assets are hardcoded but could happen right after a testnet reset.